day_two_guide/topics/restore_etcd.adoc

```diff
@@ -0,0 +1,382 @@
+
+. Check for health status (in this case, a single node):
++
+----
+# etcdctl2 cluster-health
+member 5ee217d17301 is healthy: got healthy result from https://192.168.55.8:2379
+cluster is healthy
+----
+
+. Restart the etcd service in cluster mode:
++
+----
+# rm -f /etc/systemd/system/etcd.service.d/temp.conf
+# systemctl daemon-reload
+# systemctl restart etcd
+----
+
+. Check for health status and member list
++
+----
+# etcdctl2 cluster-health
+member 5ee217d17301 is healthy: got healthy result from https://192.168.55.8:2379
+cluster is healthy
+
+# etcdctl2 member list
+5ee217d17301: name=master-0.example.com peerURLs=http://localhost:2380 clientURLs=https://192.168.55.8:2379 isLeader=true
+----
+
+. Once the first instance is running, it is safe to restore multiple etcd servers as desired.
+
+*Fix the `peerURLS` parameter*
+
+After restoring the data and creating a new cluster, the `peerURLs` parameter
+shows `localhost` instead the IP where etcd is listening for peer
+communication:
+
+----
+# etcdctl2 member list
+5ee217d17301: name=master-0.example.com peerURLs=http://*localhost*:2380 clientURLs=https://192.168.55.8:2379 isLeader=true
+----
+
+[discrete]
+== Procedure
+
+. Get the member ID from the `etcdctl member list` output.
+
+. Get the IP where etcd is listening for peer communication:
++
+----
+$ ss -l4n | grep 2380
+----
+
+. Update the member information with that IP:
++
+----
+# etcdctl2 member update 5ee217d17301 https://192.168.55.8:2380
+Updated member with ID 5ee217d17301 in cluster
+----
+
+. To verify, check that the IP is in the output of the following:
++
+----
+$ etcdctl2 member list
+5ee217d17301: name=master-0.example.com peerURLs=https://*192.168.55.8*:2380 clientURLs=https://192.168.55.8:2379 isLeader=true
+----
+
+*Add more members*
+
+In the instance joining the cluster:
+
+. Get the etcd name for the instance in the `ETCD_NAME` variable:
++
+----
+# grep ETCD_NAME /etc/etcd/etcd.conf
+----
+
+. Get the IP where etcd listens for peer communication:
++
+----
+# grep ETCD_INITIAL_ADVERTISE_PEER_URLS /etc/etcd/etcd.conf
+----
+
+. Delete the previous etcd data:
++
+----
+# rm -Rf /var/lib/etcd/*
+----
+
+. On the etcd host where etcd is properly running, add the new member:
++
+----
+$ etcdctl2 member add <name> <advertise_peer_urls>
+----
+
+. The command outputs some variables. For example:
++
+----
+ETCD_NAME="master2"
+ETCD_INITIAL_CLUSTER="master1=https://10.0.0.7:2380,master2=https://10.0.0.5:2380"
+ETCD_INITIAL_CLUSTER_STATE="existing"
+----
++
+Add those values to the `/etc/etcd/etcd.conf` file of the new host:
++
+----
+# vi /etc/etc/etcd.conf
+----
+
+. Once those values are replaced, start the etcd service in the node joining
+the cluster:
++
+----
+# systemctl start etcd.service
+----
+
+. Check for error messages:
++
+----
+$ journalctl -fu etcd.service
+----
+
+. Repeat the above for every etcd node joining the cluster.
+
+. Verify the cluster status and cluster health once all the nodes joined:
```

Comment:
. After you add all the nodes to the cluster, verify the cluster status and cluster health: